### DISCUSSION TITLE
How can I add add chemical equations to a Quarto document?

### DISCUSSION THREAD

**Discussion by murpholinox** [quarto-dev/quarto-cli#5226] [Q&A] [answered]
I have been through the whole quarto web site and cannot find something useful to add chemical equations in a quarto document. That would be nice.

A few examples like in page 4 of this great manual https://mirror2.sandyriver.net/pub/ctan/macros/latex/contrib/mhchem/mhchem.pdf

I am not sure if this is totally new or just not documented well enough.

**Comment by mcanouil** [comment]
Quarto website cannot document and showcase everything that is possible using HTML or LaTeX.

You can add extra LaTeX packages by modifying partials (<https://quarto.org/docs/journals/templates.html#latex-partials>) or by using `include-*` YAML keys (<https://quarto.org/docs/output-formats/pdf-basics.html#latex-includes>).

Same principles for HTML:
- partials (<https://quarto.org/docs/journals/templates.html#html-partials>)
- `include-*` (<https://quarto.org/docs/output-formats/html-basics.html#includes>)

For example, you can use MathJax JavaScript library (default equation environment for HTML-based format in Quarto) and the extension `mhchem`, see <https://docs.mathjax.org/en/latest/input/tex/extensions/mhchem.html>.
Regarding examples of customising MathJax, there are several examples in GitHub Discussions and for example a short blog post of mine (<https://mickael.canouil.fr/posts/2023-03-12-quarto-mathjax-packages/>).

**Comment by astoeriko** [comment, marked as answer]
I agree that some hints in the documentation would be nice since formatting chemical equations is quite a common need in scholarly writing.
As far as I see, using the mhchem package inside math delimiters (e.g., `$\ce{H2O}$`) currently is the best option. It works both for HTML (via MathJax) and pdf (via the Latex package). If you need docx output, you might try using [this mhchem lua filter](https://github.com/pandoc/lua-filters/tree/master/mhchem) for pandoc. I just recently came across it ([here](https://github.com/jgm/pandoc/issues/6668)), and did not try it myself yet, but it looks promising. Looking at the Quarto [documentation on custom filters](https://quarto.org/docs/extensions/filters.html), it should not be too complicated to use.

**Reply by baptiste** [reply]
It's quite promising that [mhchemParser](https://github.com/mhchem/mhchemParser/) is written in Typescript, and this could be combined with the [latex-environment](https://github.com/quarto-ext/latex-environment#example-1) filter to have a single notation `[H2O]{.ce}` that gets parsed and expanded into raw maths, available on all standard output formats (or passed through as `\ce{H20}` in the case of LaTeX, in case it offers more functionality).

**Reply by ghost** [reply]
> I agree that some hints in the documentation would be nice since formatting chemical equations is quite a common need in scholarly writing.

FYI @cwickham
